engine/src/main/java/org/entando/entando/aps/system/services/component/ComponentService.java

```diff
@@ -102,12 +104,17 @@ public ComponentDeleteResponse deleteInternalComponents(List<ComponentDeleteRequ
                     .ifPresentOrElse(service -> {
                         ComponentDeleteResponseRow responseRow = null;
                         try {
-                            responseRow = service.getComponentDto(code)
-                                    .map(c -> this.extractUsageDetails(code, service))
-                                    .map(usage -> check(components, code, usage))
-                                    .map(c -> delete(c, type, service))
-                                    .orElseGet(() -> ComponentDeleteResponseRow.builder().code(code).type(type)
-                                    .status(ComponentDeleteResponse.STATUS_FAILURE).build());
+                            Optional<IComponentDto> dto = service.getComponentDto(code);
+                            if (dto.isPresent()) {
+                                responseRow = dto.map(c -> this.extractUsageDetails(code, service))
+                                        .map(usage -> check(sortedComponents, code, usage))
```

Comment:
this is a confusing mix of old and new you should rewrite to make it more clear, clean, concise and readable:
```
responseRow = service.getComponentDto(code)
                                    .map(d -> deleteComponentAndBuildResponse(
                                            d, sortedComponents, code, type, service
                                    ))
                                    .orElseGet(() -> ComponentDeleteResponseRow.builder().code(code).type(type)
                                            .status(ComponentDeleteResponse.STATUS_SUCCESS).build());
-----
private ComponentDeleteResponseRow deleteComponentAndBuildResponse(IComponentDto dto,
                                                                       List<ComponentDeleteRequestRow> sortedComponents,
                                                                       String code,
                                                                       String type, IComponentUsageService service) {
        return Optional.of(dto).map(c -> this.extractUsageDetails(code, service))
                .map(usage -> check(sortedComponents, code, usage))
                .map(c -> delete(c, type, service))
                .orElseGet(() -> ComponentDeleteResponseRow.builder().code(code).type(type)
                        .status(ComponentDeleteResponse.STATUS_FAILURE).build());
    }
```

Reply:
Done, good advice